### DISCUSSION TITLE
3800mAh battery profile (enhancement)

### DISCUSSION THREAD

**Discussion by aliatx2017** [egzumer/uv-k5-firmware-custom#266] [Ideas]
Hello and thanks for your fantastic project and the time you put into it!

I know that some of us are getting uv-5r radios for its bigger 3800mAh battery. With a simple mod (break off center tab) this battery mounts and works in uv-k5(8)/k6 radios.

Would it be possible to add a 3800mAh battery profile to code/hidden menu?

Respectfully,
Alex

**Comment by gartnl** [comment]
I wonder if it is really 3800 mAh. I've got 2 5R Plus's  and both have a stock 2200 mAh battery. 
The 3800 mAh batteries are sold as replacement, but those might be Chinese milliamps ..

**Comment by pietss** [comment]
I asked a similar question some time ago
https://github.com/egzumer/uv-k5-firmware-custom/issues/130
As for the last part of the answer, as far as I'm concerned I don't have the ability, but it doesn't matter much

**Comment by egzumer** [comment]
Above answers are correct.
It is not about mAh, it's about cells chemistry. It might be the same as one of those that are already there.

These are discharge curve points (voltage, percentage):
https://github.com/egzumer/uv-k5-firmware-custom/blob/1a53586922880afbec678f3b855486918811057e/helper/battery.c#L54

**Reply by egzumer** [reply]
> As an electronics engineer and ham licensed, I’m absolutely sure the first table (1600mAh) has two mistakes: {760, 25} and {729,6}, should be {760, 75} and {729,48}, or close to 75% and 48% respectively... Maybe you tested a faulty 1600?!

Well I'm also an electrical engineer, but I measured the discharge curve, did you?

**Reply by egzumer** [reply]
> “It is not about mAh, it's about cells chemistry”, you’re right, 
...
So… I can’t understand where did the different discharge voltages-capacity values in the tables came from?!

You started with an answer for a question asked at the end of you paragraph.

> After two days of strugling to understand why my K5(8) + 1600mAh battery had an abrupt drop of voltage bars (but not voltage), I arrived to this page. 

Does it show wrong percentage values? I.e. the radio works for a long, long time after it once showed 6%? Or you just think it should be higher for a given voltage?

> Now my mistery is solved, should use the 2200 profile instead of 1600, that one is correct.

You can use whichever you want, but you should probably use 1600 curve.

> Maybe you tested a faulty 1600?!

I don't think so, it works perfectly fine, the test ended with the expected capacity. Nobody ever complained about wrong percentage values for 1600 battery curve.

**Reply by egzumer** [reply]
> If using 1600 calibration, it gets fast from 4 to 1 bar (a few hours),

I was asking about percentage, you are telling me about bars. Bars disappear at certain percentage values which are mathematical rounding to base 4 (or something like that because there is 0 bars, and also 0 bars blinking at 5%). It seemed to be wrong for me too, but it is correct, it is hard to show accurate state of charge on 4 bars, and stock firmware bar indicator was crap too either way. I'm not saying the curve is correct for your battery, but the bars thing have it's logic that you may find odd. The percentage is what is relevant.

> Lets look at the 2 tables and the graph above

It is not a graph of a quansheng battery, is it? Liion batteries use different chemistries (I pointed that before and you keep ignoring that) and will have different discharge curves. The curve is very flat for the most part and can give huge errors in state of charge value if you use some random curve found on the internet.
1600 uses pouch prismatic cells, 2200 uses cylindrical 18650 cells. I did discharge curves for both and this is what I got. Maybe quansheng changed supplier and you have different cells. 1600 works for me perfectly, if 2200 curve works better for you, then use that one.

> vloggers, that had difficulties to understand the menus(!)

Different radios work differently, have different menus. I have problems navigating yeasu radios, life. Quansheng works like it works, this firmware is an expansion on stock firmware, and it is not going to change much more. The voltage has to be calibrated because it is not calibrated properly in the factory. It is what it is.

**Comment by AndrewTXJ** [comment]
maybe just try to look for the voltage, it would be easier, i don't have the bigger battery, i have the stock one, which is 1600mAh, i always look for the voltage, if it drop down below 7.5V that is time to charge, if it is 8.5v and above, it is full or 99% full, if you get a bigger battery, it is the same.

**Comment by BruMcD** [comment]
![Screenshot 2024-03-18 152449](https://github.com/egzumer/uv-k5-firmware-custom/assets/157917555/68882570-8277-4176-85e7-5764ac9021da)
A couple of notes:
The stock 1600 battery has a higher than average mid point voltage for a 7.2V (sic) unit ,   By 7.2V it is pretty much empty  
Not so the '2200's' - very different curves are needed and so they are provided

.. also the Radios need the battery calibration setting corrected in the hidden menu using a **highly accurate multimeter** and ideally around the actual mid point of discharge (where remaining 'battery %' errors would be greatest) before doing much else

Having run tests on 8  1600mAh's and logged displayed %  vs that point in that battery/set run's total time to exhausted I can say that batteries vary - but there is very little if anything wrong with Egzumer's 1600mA battery meter % or Icons 

And yes, I'm an over qualified engineer too  😂😇

**Reply by egzumer** [reply]
Looking at the graph the percentage calculation could be improoved just a bit, but damn, it is pretty good as it is. I must say, I'm proud of it 🤓🤣

That's case closed, thanks 🤠
